src/process/message_processor.go

```diff
@@ -1,48 +1,94 @@
 package process
 
 import (
+	"BGPAlert/analyze"
 	"BGPAlert/common"
 	"errors"
 	"fmt"
 	"time"
 )
 
-// Constantly reads channel of messages and stores them in Window objects to send through windowChannel for analysis
-func ProcessBGPMessages(msgChannel chan common.BGPMessage, windowChannel chan common.Window) error {
-	var bucketMap = make(map[time.Time][]common.BGPMessage)
+// Constantly read messages from channel, build up windows with frequency maps, calling analysis on windows when full
+func ProcessBGPMessages(msgChannel chan common.BGPMessage) error {
+	maximumBuckets := 43
+	// Each bucket is 60s
+	parseDuration, err := time.ParseDuration("60s")
```

Comment:
Can't you do something like `60 * time.Second` instead of parsing here? In that case, you'd be sure that the computation never gives you an error. (I don't remember my Go tho!)

Reply:
The function ParseDuration only takes in a string so I don't think there's an easy way around it.